### PR TITLE
[Merged by Bors] - chore: download recent curl if necessary on x64 Linux

### PR DESCRIPTION
**Pull request by Kha**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Cache/IO.lean [hunk 1]
```diff
@@ -77,15 +84,30 @@ def runCmd (cmd : String) (args : Array String) (throwFailure := true) : IO Stri
...
+      let _ := @leOfOrd
+      if version >= (7, 81) then return true
+      -- TODO: support more platforms if the need arises
+      if (← runCmd "uname" #["-ms"] false).trim == "Linux x86_64" then
+        IO.println s!"curl is too old; downloading more recent version"
+        IO.FS.createDirAll IO.CACHEDIR
+        -- NOTE: we host only one version of curl, which we assume to be at least as recent
+        -- as the version encoded in `CURLBIN`
+        let _ ← runCmd "curl" #[
+          "https://pp.ipd.kit.edu/~ullrich/tmp/curl-x86_64-linux-static", "-o", CURLBIN.toString]
```
Comment (Kha) [inline Cache/IO.lean:106]: Obviously this should be hosted somewhere else, could someone move it to the Azure cache?

Comment (gebner) [reply Cache/IO.lean:107]: Should we go with the time-honored tradition of abusing github releases for this?  I can make a leanprover-community/static-curl repo and give you access if you want.

Comment (Kha) [reply Cache/IO.lean:107]: Sounds fine too, whatever is simpler I would say

Comment (gebner) [reply Cache/IO.lean:107]: Invite sent!

Comment (Kha) [reply Cache/IO.lean]: Done, and added aarch64 for good measure

File: Cache/IO.lean [hunk 2]
```diff
@@ -43,9 +43,20 @@ initialize CACHEDIR : FilePath ← do
 def CURLCFG :=
   IO.CACHEDIR / "curl.cfg"
 
+/-- curl version at https://github.com/leanprover-community/static-curl -/
+def CURLVERSION :=
+  "7.88.1"
```
Comment (Kha) [inline Cache/IO.lean:48]: Why are we screaming again?

**Comment by Kha** [comment]
By the way, I'm surprised we haven't hit https://github.com/leanprover/elan/commit/342a0ca3870d06bd32db4a16b1549219657b5d55 yet

**Review by arthurpaulino — COMMENTED** [review]
There's a bad interaction with the `clean` and `clean!` commands, which can delete the downloaded `curl`. Those should then skip `curl` for deletion, or something like that

**Comment by Kha** [comment]
No, they only remove `*.gz` files

**Comment by arthurpaulino** [comment]
Right, bad memory 🤦🏼

**Review by arthurpaulino — APPROVED** [review]

**Pull request closed without merging**